Aplib.Core/Logging/LogNode.cs

```diff
@@ -0,0 +1,42 @@
+using System.Collections.Generic;
+
+namespace Aplib.Core.Logging
+{
+    /// <summary>
+    /// Represents a node in the log tree.
+    /// </summary>
+    public class LogNode
+    {
+        /// <summary>
+        /// The loggable object of the node.
+        /// </summary>
+        public ILoggable Loggable { get; }
+
+        /// <summary>
+        /// The depth at which this node resides.
+        /// </summary>
+        public int Depth { get; }
+
+        /// <summary>
+        /// The children of the node.
+        /// </summary>
+        public List<LogNode> Children { get; }
+
+        /// <summary>
+        /// Initialize a new <see cref="LogNode" /> from a given loggable object,
+        /// the depth, and optionally a list of children.
+        /// </summary>
+        /// <param name="loggable">The loggable object of the node.</param>
+        /// <param name="depth">The depth at which this node resides.</param>
+        /// <param name="children">The children of the node. If omitted, an empty list will be used.</param>
+        public LogNode(ILoggable loggable, int depth, List<LogNode> children)
```

Comment:
Are these children with correct depth etc.? If that is assumed, could you add that to the docs? If not, could you set the depth for the children or omit this constructor?

Reply:
Should be children with the correct depth yes, is assumed. Will add